### DISCUSSION TITLE
Voting on games - Review

### DISCUSSION THREAD

**Discussion by nguillaumin** [atari-legend/atari-legend#172] [Show and tell] [answered]
Hi all, @stgraveyard ,

I have implemented basic voting system for games. There's a new section on the game info card where one can see the score of a game, the vote distribution, and cast a vote:

![vote](https://user-images.githubusercontent.com/762512/151622395-f3caddd0-e20b-46f6-a468-ac31a601e26b.png)

One needs to be logged in to vote, and can only vote once per game. The score has 5 steps from bad to good. The little graphic shows the vote distribution (how many voted 0, how many voted 1, how many 2, etc.) and there's also the number of votes.

I also added a "Top 5 games"  in the "Tops" card, on the game page:

![to](https://user-images.githubusercontent.com/762512/151622926-e347ce71-69e7-4e05-9a58-3ee25b6e4763.png)

It's deployed on the dev system. I also inserted random votes on Epic just for testing: https://dev.atarilegend.com/games/1668

I would like to extend it later to have a "top worst 5 games", and also shows who voted what for each game, and possibly more. However we first need to collect some votes to make it useful, so I'd like to deploy it to prod and let people vote for a few weeks/months.

Let me know what you think?

Nico

**Comment by stgraveyard** [comment]
Hi Nico,

I think it is a very classy looking implementation, wow! 

I have tested in all resolution, but I need to say this (maybe just to show you how autistic I can be) - Look at the button : 

![image](https://user-images.githubusercontent.com/4339294/151632641-6a4a66c1-fbbb-40f1-9245-e9196a6939af.png)

And look now : 

![image](https://user-images.githubusercontent.com/4339294/151632687-24ba8d5c-8824-4e53-bdb7-ca0a9cceb082.png)

This is in fullscreen here. Nico, for me it is ok, put it in prod. It looks great. I just wanted to show you, in case you would have a quick fix for this minor thing ... 

I like the tops tile. Maybe another one can be created with a games top 5 and the graphic of the nr 1 game. Just an idea, to add a bit more color to the page. 

Also, might it be better to change the font color of the Top Individuals, Top Genres, Top Publishers ... a little? 
![image](https://user-images.githubusercontent.com/4339294/151633092-7e667a8f-cb14-433d-bda2-035857d64ace.png)
It is just at first sight, the 'Top Inviduals' is 'lost' in the '108 games'.

Nico, these are details, it looks great and I'm eternally grateful! 

Maarten

**Reply by nguillaumin** [reply]
Thanks! Always the eagle eye for details :+1:  Good ideas about adding screenshots. I have added a separate "top games" card, took some inspiration from our friends at Lemon Amiga ;)

I moved the rest below the search form:

![votes](https://user-images.githubusercontent.com/762512/151658998-3b87db5e-9568-4076-9d2e-c8db62fe54a3.png)

It's deployed on dev.

Nico

**Comment by stgraveyard** [comment]
Love the graphics. Few remarks, just something that I noticed right away : 

![image](https://user-images.githubusercontent.com/4339294/151677775-e6f3b269-3ba6-44f6-9ec4-50d7ca4eb112.png)

I think the alignment of everything should be in the middle vertically. Look at nr 3. You also see it in the screenshot of every nr. In your screenshot it looks perfect. 

![image](https://user-images.githubusercontent.com/4339294/151677812-ab0a820a-9200-4300-b3e8-ac65e429b4cf.png)

Nico, this is probably impossible, but maybe. Could you try aligning the white text here to the right and leave the blue text left? These are probably not columns so not possible I guess.

**Reply by nguillaumin** [reply]
Thanks for the suggestions! Rows were looking good on my screen but indeed in smaller resolutions they were not aligned, I fixed it. I also reformatted the "tops" card with tables and it looks better now:

![tops](https://user-images.githubusercontent.com/762512/151696510-82ffa781-ffb8-4d00-897a-dd75c00f6877.png)

It's updated on dev.

**Reply by stgraveyard** [reply]
Beautiful work. Move this to prod! ;-)

**Reply by nguillaumin** [reply]
Done! Please go and vote and spread the word :smiley:

**Reply by stgraveyard** [reply]
Perfect I will make a post on monday on social media.

**Reply by stgraveyard** [reply]
> Perfect I will make a post on monday on social media.

No, tonight when I am back home. Should be doable.

**Reply by stgraveyard** [reply]
Before I announce, I think we have a bug : 
![image](https://user-images.githubusercontent.com/4339294/151710133-78adc353-e631-4e1a-ae42-bb2e3766b628.png)

So, Pang has now 2 votes, both awesome, simcity only 1 vote, yet Simcity stays at nr 1? 

Is that normal?

**Reply by nguillaumin** [reply]
I guess yes, because what matters is the score, not the numbers of votes? They're both rated 5.00, so ex aequo.

I have added a second sort criteria so that if they have the same score, the one with the most vote wins.

**Comment by stgraveyard** [comment]
Hi Nico, 

I'm a guy who thinks a lot of things are awesome when it comes to Atari ST ;-) 

![image](https://user-images.githubusercontent.com/4339294/151751043-862eb00e-400d-4a5c-ac61-d38b6d83e516.png)

But the fact that it always creates a new line in the layout is starting to annoy me. I wonder, wouldn't it be better to just display 'Awesome' without the 'your rating'? This way the layout stays perfect. Or instead of 'your rating' use 'rated' ... I don't know. 

Cheers,
Maarten

**Reply by nguillaumin** [reply]
There's unfortunately this awkward screen size where it's not small enough to break into 2 columns instead of three, but not large enough for the small columns (left and right) to be wide enough.

Anyway I made some tweaks and only display "Your rating: ?" when there's no rating yet, otherwise just the rating name.

**Reply by stgraveyard** [reply]
This is much better to me Nico, thank you. Just to be clear, my screenshot was in 3 column mode fullscreen on my laptop (1920*1080), that is why I asked for a change. I do understand it can't be correct in every screen width, but when I watch it in fullscreen, It needs to be.

**Reply by nguillaumin** [reply]
Completely agree, however for me it was fine at exactly the same resolution ☺️ I suspect you have a browser zoom, or the display settings of Windows are set to 125% or 150% for hdpi? So there's more than one definition of "full screen" 😉

**Reply by stgraveyard** [reply]
hmmm. Browser is set to 100%. windows I don't really know tbh. It looks normal to me :-)

**Comment by nguillaumin** [comment, marked as answer]
Rating is now deployed on production.

**Reply by stgraveyard** [reply]
Am I missing something here? This has been deployed months ago, no?

**Reply by nguillaumin** [reply]
Yes, it was just to close the discussion
